### PR TITLE
added models

### PR DESCRIPTION
**Pull request by MadikhanSadykov**
I've created models according to the erd db schema
erd_v1

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    <artifactId>social-network-java</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <name>social-network-java</name>
+    <description>social-network-java</description>
+    <properties>
+        <java.version>11</java.version>
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
```
Comment (MadikhanSadykov) [inline pom.xml:21]: I'll remove these dependencies (spring data jpa, spring boot, data rest) since I'm not allowed to use them

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
```
Comment (aleksei-pronichev) [inline pom.xml:22]: delete at start

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+                <configuration>
+                    <excludes>
```
Comment (aleksei-pronichev) [inline pom.xml:65]: why?

File: src/main/java/com/madikhan/app/SocialNetworkJavaApplication.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.madikhan.app;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
```
Comment (aleksei-pronichev) [inline src/main/java/com/madikhan/app/SocialNetworkJavaApplication.java:6]: remove

File: src/main/java/com/madikhan/app/model/AuditableEntity.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+import javax.persistence.Column;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.PrePersist;
+import javax.persistence.PreUpdate;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+
+@Data
+@EqualsAndHashCode(callSuper=true)
```
Comment (aleksei-pronichev) [inline src/main/java/com/madikhan/app/model/AuditableEntity.java:17]: for what?

File: src/main/java/com/madikhan/app/model/BaseEntity.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+import java.io.Serializable;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@MappedSuperclass
+public abstract class BaseEntity<T extends Serializable> {
```
Comment (aleksei-pronichev) [inline src/main/java/com/madikhan/app/model/BaseEntity.java:19]: remove it

File: src/main/java/com/madikhan/app/model/Comment.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Data
+@EqualsAndHashCode(callSuper=true)
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Entity
+@Table(name = "comment", schema = "social-network")
```
Comment (aleksei-pronichev) [inline src/main/java/com/madikhan/app/model/Comment.java:22]: schema  ?

File: src/test/java/com/madikhan/app/SocialNetworkJavaApplicationTests.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.madikhan.app;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
```
Comment (aleksei-pronichev) [inline src/test/java/com/madikhan/app/SocialNetworkJavaApplicationTests.java:6]: remove

File: src/main/java/com/madikhan/app/util/converter/BirthdayConverter.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import java.sql.Date;
+import java.util.Optional;
+
+public class BirthdayConverter implements AttributeConverter<Birthday, Date> {
+    @Override
+    public Date convertToDatabaseColumn(Birthday birthdayEntity) {
+        return Optional.ofNullable(birthdayEntity)
+                .map(Birthday::getBirthDate)
+                .map(Date::valueOf)
+                .orElse(null);
```
Comment (aleksei-pronichev) [inline src/main/java/com/madikhan/app/util/converter/BirthdayConverter.java:15]: it's really bad practice to return null, worse only return null in Optional.orElseGet

**Pull request closed without merging**